.github/workflows/pytest.yaml

```
@@ -43,7 +43,36 @@ jobs:
             package_list.txt
             pip_list.txt
 
+      - name: Clone caret_analyze_cpp_impl
+        run: |
+          git clone https://github.com/tier4/caret_analyze_cpp_impl.git
+
+      - name: Restore Cache
+        id: cpp_impl-cache
+        uses: actions/cache/restore@v3
+        with:
+          path: caret_analyze_cpp_impl/install
+          key: ${{ runner.os }}-node-cpp_impl-${{ hashFiles('caret_analyze_cpp_impl/CARET_analyze_cpp_impl/**') }}
+
+      - name: Build caret_analyze_cpp_impl
+        if: steps.cpp_impl-cache.outputs.cache-hit != 'true'
+        run: |
+          source /opt/ros/humble/setup.bash
+          cd caret_analyze_cpp_impl/
+          colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
+        shell: bash
+
+      - name: Cache caret_analyze_cpp_impl
+        if: steps.cpp_impl-cache.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v3
+        with:
+          path: caret_analyze_cpp_impl/install
+          key: ${{ runner.os }}-node-cpp_impl-${{ hashFiles('caret_analyze_cpp_impl/CARET_analyze_cpp_impl/**') }}
+
       - name: Run pytest
         run: |
+          source caret_analyze_cpp_impl/install/setup.bash
           . /opt/ros/humble/setup.sh
```

```suggestion
          . /opt/ros/humble/setup.sh
          source caret_analyze_cpp_impl/install/setup.bash
```

This might be a good order to source them.

---

I merged it into the previous commit since it was a minor fix.